### ISSUE TITLE
bezier subdivisions algorithms

### ISSUE THREAD

**Issue by tailangjun** [memononen/nanovg#381]
wac__tesselateBezier Recursion termination when (d2 + d3)*(d2 + d3) < engine->tessTol * (dx*dx + dy*dy). I know d2 and d3 both are area, using cross product. Do you mean the condition is the edges are too short or the angle is too small?

**Comment by memononen** [comment]
The algorithm is adapted from here: http://antigrain.com/research/adaptive_bezier/

**Comment by tailangjun** [comment]
I understand tesselateBezier()  algorithms now. Another question is QuadTo() why not use midpoint.
```
    float vals[] = { WAC_COMMANDS_BEZIERTO,
        0.5f*(x0 + cx), 0.5f*(y0 + cy),
        0.5f*(cx + x), 0.5f*(cy + y),
        x, y };
```

**Comment by memononen** [comment]
I don't know what you imply by using midpoint. Is there another algorithm that converts quadric bezier to cubic by using a midpoint?

**Comment by tailangjun** [comment]
you are right the algorithm: [http://fontforge.github.io/bezier.html](url)
